src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.macios.cs

```diff
@@ -66,9 +66,9 @@ public override void ViewWillDisappear(bool animated)
 	[MemberNotNull(nameof(VirtualView), nameof(ViewController))]
 	public void SetElement(IPopup element)
 	{
-		if (element.Parent?.Handler is not PageHandler)
+		if (element.Parent?.Handler is null)
```

Comment:
Our popup relies on `PageHandler` or `Page`, so we can't use the `null` check here, because in this case, we will allow devs to pass an `EntryHandler` for example.

Reply:
```suggestion
		if (element.Parent?.Handler is not PageHandler && element.Parent?.Handler is not UIViewController)
		{
			throw new InvalidOperationException($"The {nameof(element.Parent)} must be of type {typeof(PageHandler)} or {typeof(UIViewController)}.");
		}
```
Please let me know if it is correct. After your opinion, will be pushing the code.

Reply:
I'll try here and see how that goes. Thanks